src/main/java/org/folio/ed/service/SecurityManagerService.java

```diff
@@ -0,0 +1,205 @@
+      var userId = createFolioUser(username);
+      saveCredentials(systemUserParameters);
+      assignPermissions(userId);
+    }
+
+    var backgroundUserApiKey = loginSystemUser(systemUserParameters);
+    systemUserParameters.setOkapiToken(backgroundUserApiKey);
+    saveSystemUserParameters(systemUserParameters);
+  }
+
+  private String loginSystemUser(SystemUserParameters params) {
+    var response = authnClient.getApiKey(params);
+    var headers = response.getHeaders().get(XOkapiHeaders.TOKEN);
+    if (CollectionUtils.isEmpty(headers)) {
+      throw new IllegalStateException(String.format("User [%s] cannot log in", params.getUsername()));
+    } else {
+      return headers.get(0);
+    }
+  }
+
+  private SystemUserParameters buildDefaultSystemUserParameters(String username, String password, String okapiUrl, String tenantId) {
+    return SystemUserParameters.builder()
+      .id(UUID.randomUUID())
+      .username(username)
+      .password(password)
+      .okapiUrl(okapiUrl)
+      .tenantId(tenantId).build();
+  }
+
+  @CachePut(value = "systemUserParameters", key="#systemUserParams.tenantId")
+  public void saveSystemUserParameters(SystemUserParameters systemUserParams) {
+    systemUserParametersRepository.save(systemUserParams);
+  }
+
+  @Cacheable(value = "systemUserParameters", key="#tenantId")
+  public SystemUserParameters getSystemUserParameters(String tenantId) {
+    final String sqlQuery = "SELECT * FROM " + moduleMetadata.getDBSchemaName(tenantId) + ".system_user_parameters";
+    var query = em.createNativeQuery(sqlQuery, SystemUserParameters.class); //NOSONAR
+    return (SystemUserParameters) query.getSingleResult();
+  }
+
+  private Optional<User> getFolioUser(String username) {
+    var query = "username==" + username;
+    var results = usersClient.query(query);
+    return results.getResult().stream().findFirst();
+  }
+
+  private String createFolioUser(String username) {
+    final var user = createUserObject(username);
+    final var id = user.getId();
+    usersClient.saveUser(user);
+    return id;
+  }
+
+  private void updateUser(User existingUser) {
+    log.info("Have to update  user [{}]", existingUser.getUsername());
+    if (existingUserUpToDate(existingUser)) {
+      log.info("The user [{}] is up to date", existingUser.getUsername());
+    }
+    usersClient.updateUser(existingUser.getId(), populateMissingUserProperties(existingUser));
+    log.info("Update the user [{}]", existingUser.getId());
+  }
+
+  private void saveCredentials(SystemUserParameters systemUserParameters) {
+
+    authnClient.saveCredentials(systemUserParameters);
+
+    log.info("Saved credentials for user: [{}]", systemUserParameters.getUsername());
+  }
+
+  private boolean assignPermissions(String userId) {
+    List<String> perms = readPermissionsFromResource(PERMISSIONS_FILE_PATH);
+
+    if (isEmpty(perms)) {
+      throw new IllegalStateException("No permissions found to assign to user with id: " + userId);
```

Comment:
Maybe it is better to return false here, as in opposite side assignPermissions() all times returns true. Or make it void.